src/libsync/abstractnetworkjob.cpp

```diff
@@ -179,21 +160,32 @@ void AbstractNetworkJob::sendRequest(const QByteArray &verb,
     }
     auto reply = _account->sendRawRequest(verb, _request.url(), _request, requestBody);
     if (_requestBody) {
-        _requestBody->setParent(reply);
+        _requestBody->setParent(this);
```

Comment:
Do we know about the side effects of this change now?

Reply:
The body will only be deleted once the job is deleted.
As we might repeatedly call sendRequest (retry) this should be more correct.